world/handlers.go

```diff
@@ -176,6 +182,26 @@ func (server *Server) handlePartyEvent(conn mnet.Server, reader mpacket.Reader)
 		p.WriteBytes(reader.GetBuffer())
 		server.channelBroadcast(p)
 	default:
-		log.Println("Unkown party event type:", op)
+		log.Println("Unknown party event type:", op)
+	}
+}
+
+func (server *Server) handleChangeRate(conn mnet.Server, reader mpacket.Reader) {
```

Comment:
This function should also detect if the rates have gone above `2x` to inform the login server so that it can set the relevant information on world select as I think you can state there is an exp event going on.

https://github.com/Hucaru/Valhalla/blob/master/login/packets.go#L203

You will need to add this to https://github.com/Hucaru/Valhalla/blob/bab971b99f24e989f401d2017a8d5e54d34e301f/internal/serialisable_data.go#L9 
and
 https://github.com/Hucaru/Valhalla/blob/bab971b99f24e989f401d2017a8d5e54d34e301f/internal/serialisable_data.go#L17

Reply:
rates on a private server will probably always be higher then the normal rates. do we want to inform the client like its an exp event?

Reply:
Whoa, new updates. It's also nice to see a new contributor. :)

May I suggest if rates go higher than `1x`? Exp/Drop events were rare occurrences that happened only a handful of times throughout a year in GMS. I doubt many would play a private server at this low a rate, but it would be nice to see this source as close to the original game.

However, I'm assuming the mentioned function above could just as easily have its value changed to what the server owner considers as the default rates, and anything higher would be considered an Exp/Drop event? If that is the case, then I guess my `1x` suggestion wouldn't matter.

Reply:
I agree with @LuneArk's last paragraph: the exp rates should be set via a world config file which then distributes it to the channels. Then anything that is set above that value triggers an exp event notification back to the login server so that the event can be displayed on world select

Reply:
Got you!